### PR TITLE
Remove all references to NationalPACSAccessionNumber

### LINKED ISSUE #804: Generalise image ingest details
- Directory scan messages refer to `NationalPACSAccessionNumber`, when this isn't always relevant
- MongoDB document header also contains this. Is it used at all?

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Removes all uses of `NationalPACSAccessionNumber` in the codebase. This is currently stored in the message headers during the data load process, and also in the MongoDB document headers. The value is never actually read from though, and so it is safe to remove. Documents in MongoDB which already have this value set will still work with the reprocessor, since it will simply ignore the value. The value can always be retrieved from each `DicomDataset` as standard.

This allows generalisation of the messages and MongoDB document schema to generalise to datasets not in the EPCC PACS format.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [x] Removal / depreciation
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [ ] Updated any relevant API documentation
- [x] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues

- Closes #804

### CODE REVIEW

**Comment by rkm** [comment]
> This pull request **introduces 1 alert** when merging [f8ae735](https://github.com/SMI/SmiServices/commit/f8ae73514333ee1525f187ab2ebc3c6152aa7102) into [29b975f](https://github.com/SMI/SmiServices/commit/29b975f11ced19f9d094c6a52ceea8e9f362331e) - [view on LGTM.com](https://lgtm.com/projects/g/SMI/SmiServices/rev/pr-93ded08f2f27ab20b03675477799ded186135c06)
> 
> **new alerts:**
> 
> * 1 for Constant condition

Go home, you're drunk!

File: src/common/Smi.Common/Messages/DicomFileMessage.cs [hunk 1]
```diff
@@ -119,8 +111,7 @@ public override string ToString()
 
         private bool Equals(DicomFileMessage other)
         {
-            return
-                string.Equals(NationalPACSAccessionNumber, other.NationalPACSAccessionNumber)
+            return true
```
Comment (tznind) [inline src/common/Smi.Common/Messages/DicomFileMessage.cs:114]: why "true" why not just remove the leading &&

**Review by tznind — APPROVED** [review]

File: src/common/Smi.Common/Messages/DicomFileMessage.cs (hunk 1 above)
Comment (tznind) [reply src/common/Smi.Common/Messages/DicomFileMessage.cs:114]: This is also the '1 for Constant condition' lgtm issue

Comment (rkm) [reply src/common/Smi.Common/Messages/DicomFileMessage.cs:114]: It's an Incredibly minor style choice, so all the conditionals on the following lines line up nicely. I.e.

```csharp
return
    Foo()
    && Bar()
    ...;
```

or

```csharp
return Foo()
    && Bar()
    ...;
```

compared to

```csharp
return true
    && Foo()
    && Bar()
    ...;
```

Comment (tznind) [reply src/common/Smi.Common/Messages/DicomFileMessage.cs:114]: Ok... I would just use whitespace personally.  The true technically introcues 1 more machine OP although the precompiler probably refactores it away during building.  But it's your code.

```
return
       Foo()
    && Bar()
    ...;
```

**Pull request merged**